guide/sections/part2/global-services.adoc

```diff
@@ -218,3 +222,38 @@ wis2-gdc is managed as a free and open source project.  Source code, issue track
+
+===== Technical considerations
+
+* The Global Replay provides Global Services and Data Consumers with a mechanism to search and query for notification messages of interest.
+* The Global Discovery Catalogue implements the OGC API – Features – Part 1: Core standardfootnote:[OGC-API Features - Part 1 TODO ADD LINK], adhering to the following conform    ance classes and their dependencies.
+** TODO: add Requirements classes
+** JSON (Building Block)
+* A Global Replay shall subscribe to the topics `+origin/a/wis2/#+` and `+cache/a/wis2/#+`.
+* The Global Replay will make notification messages available via the collectio identifier of `wis2-notification-messages`.
+* A single Global Replay instance is sufficient for WIS2.
+* Multiple Global Replay instances may be deployed for resilience.
+* Global Replay instances operate independently of each other; each Global Replay instance will hold notification messages according to the required retention period.  Global Replays do not need to synchronise between themselves.
+* A Global Replay is populated with notification messages from a Global Broker instance.
+* A Global Replay should connect and subscribe to more than one Global Broker instance to ensure that no messages are lost in the event of a Global Broker failure. A Global Replay instance will discard duplicate messages as needed.
+* A Global Replay will validate notification messages against the WIS Notification Message (WNM).  Valid WIS Notification Messages will be ingested into the index.  Invalid or malformed notification messages will be discarded. TODO should we validate WNM? or moot point?validate
+* A Global Replay will add a property called ``properties.topic`` to identify the topic from the which the notification message was published.
+* A Global Replay will remove notification messages after the required retention period.
+* As a convention Global Replay centre-id will be ``tld-{centre-name}-global-replay``.
+
+===== Global Replay reference implementation: wis2-grep
+
+To provide a Global Replay, members may use whichever software components they consider most appropriate to comply with WIS2 Technical Regulations.
+
+To assist Members participation in WIS2, a free and open-source Global Replay Reference Implementation is made available for download and use.  wis2-grep builds on mature and robust free and open-source software components that are widely adopted for operational use.
+
+wis2-grep provides functionality required for the Global Replay, providing the following technical functions:
+
+* notification messages subscription and publication from the Global Broker
+* notification message validation, ingest and publication (TODO: should we validate or not?)
+* WNM compliance (TODO: should we validate or not?)
+* OGC API - Features - Part 1: Core compliance
+
+wis2-grep is managed as a free and open source project.  Source code, issue tracking and discussions are hosted in the open on GitHub: https://github.com/wmo-im/wis2-grep.
```

Comment:
It doesn't say how it serves the data (aka here Notification Messages) to users.

Reply:
By stating compliance to OGC API - Features - Part 1: Core, this means HTTP API workflow.